submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/submit/SubmissionDraftApiTest.kt

```diff
@@ -110,43 +111,45 @@ class SubmissionDraftApiTest(
     }
 
     @Test
-    fun `12-6 delete a draft directly`() {
-        val pageTab =
-            jsonObj {
-                "accno" to "ABC-128"
-                "type" to "Study"
-            }.toString()
-        webClient.submit(pageTab, JSON)
-
-        webClient.deleteSubmissionDraft("ABC-128")
-
-        assertThat(dataService.getUserData(SuperUser.email, "ABC-128")).isNull()
-    }
+    fun `12-6 delete a draft directly`() =
+        runTest {
+            val pageTab =
+                jsonObj {
+                    "accno" to "ABC-128"
+                    "type" to "Study"
+                }.toString()
+            webClient.submit(pageTab, JSON)
+
+            webClient.deleteSubmissionDraft("ABC-128")
+
+            assertThat(draftPersistenceService.findSubmissionDraft(SuperUser.email, "ABC-128")).isNull()
+        }
 
     @Test
-    fun `12-7 re submit from draft`() {
-        webClient.submit(
-            jsonObj {
-                "accno" to "ABC-129"
-                "type" to "Study"
-            }.toString(),
-            JSON,
-        )
-
-        webClient.updateSubmissionDraft(
-            "ABC-129",
-            jsonObj {
-                "accno" to "ABC-129"
-                "ReleaseDate" to "2021-09-21"
-                "type" to "Study"
-            }.toString(),
-        )
-
-        webClient.submitFromDraft("ABC-129")
-
-        assertThat(dataService.getUserData(SuperUser.email, "ABC-129")).isNull()
-        assertThat(dataService.getUserData(SuperUser.email, "ABC-129")).isNull()
-    }
+    fun `12-7 re submit from draft`() =
+        runTest {
+            webClient.submit(
+                jsonObj {
+                    "accno" to "ABC-129"
+                    "type" to "Study"
+                }.toString(),
+                JSON,
+            )
+
+            webClient.updateSubmissionDraft(
+                "ABC-129",
+                jsonObj {
+                    "accno" to "ABC-129"
+                    "ReleaseDate" to "2021-09-21"
+                    "type" to "Study"
+                }.toString(),
+            )
+
+            webClient.submitFromDraft("ABC-129")
+
+            assertThat(draftPersistenceService.findSubmissionDraft(SuperUser.email, "ABC-129")).isNull()
+            assertThat(draftPersistenceService.findSubmissionDraft(SuperUser.email, "ABC-129")).isNull()
```

Comment:
it looks to be we need to check submission is there with the given data, dontt we?